packages/fxa-admin-panel/src/components/PageAccountDelete/index.tsx

```diff
@@ -0,0 +1,143 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useMutation, useQuery } from '@apollo/client';
+import React, { useState } from 'react';
+import {
+  ACCOUNT_DELETE_MUTATION,
+  ACCOUNT_DELETE_TASK_STATUS_QUERY,
+  AccountDeleteResult,
+  TaskStatus,
+} from './index.gql';
+
+export const AccountDeleteResults = ({
+  list,
+}: {
+  list: Array<AccountDeleteResult>;
+}) => {
+  const statusResult = useQuery(ACCOUNT_DELETE_TASK_STATUS_QUERY, {
+    variables: {
+      taskNames: list.filter((x) => x.taskName).map((x) => x.taskName),
+    },
+    pollInterval: 5000,
+  });
+
+  if (statusResult.loading) {
+    return <>Loading...</>;
+  }
+
+  const rows = list.map((row) => (
+    <tr>
+      <td>{row.locator}</td>
+      <td>
+        {statusResult.data?.getDeleteStatus?.find(
+          (x: TaskStatus) => row.taskName === x.taskName
+        )?.status || row.status}
+      </td>
+    </tr>
+  ));
+
+  return (
+    <>
+      <table cellPadding={12}>
+        <tr>
+          <th>Account</th>
+          <th>Current Status</th>
+        </tr>
+        {rows}
+      </table>
+    </>
+  );
+};
+
+export const PageAccountDelete = () => {
+  const [inProgress, setInProgress] = useState(false);
+  const [results, setResults] = useState<Array<AccountDeleteResult>>([]);
+  const [deleteAccountMutation] = useMutation(ACCOUNT_DELETE_MUTATION);
+
+  const saveButtonClass = () => {
+    const base =
+      'bg-grey-10 border-2 p-1 border-grey-100 font-small leading-6 rounded';
+    const active =
+      'text-red-700 hover:text-red-700 hover:border-2 hover:border-grey-10 hover:bg-grey-50';
+    const inactive = 'text-grey-700 cursor-not-allowed';
+    return `${base} ${!inProgress ? active : inactive}`;
+  };
+
+  const handleSubmit = async (e: React.SyntheticEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setInProgress(true);
+    setResults([]);
+
+    try {
+      const formData = new FormData(e.currentTarget);
+      const formJson = Object.fromEntries(formData.entries());
+      const list = formJson.accountList
+        .toString()
+        .split(/\n|,/)
+        .map((x) => x.trim())
+        .filter((x) => x.length > 0);
+
+      if (list.length > 1000) {
+        alert('Enter fewer than 1000 emails or uids!');
+        setResults([]);
+        return;
+      }
+
+      const result = await deleteAccountMutation({
+        variables: { locators: list },
+      });
+      setResults(result.data?.deleteAccounts || []);
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setInProgress(false);
+    }
+  };
+
+  return (
+    <>
+      <h2 className="header-page">Delete Accounts</h2>
+      <p>
+        This page allows you to delete user accounts.{' '}
+        <b>Be careful! Data will be lost through this action!</b>
+      </p>
+      <br />
+      <p>
+        Provide a list of up to 1000 emails or UIDs separated by either a coma
```

Comment:
Was the 1000 based on the delete script? It does seem kind of high imo

Reply:
This is just a ball park estimate that could be quite off. There was no actual upper bound. I didn't want to totally clobber the server though, so put some limit in place.